riscv-page-tables/src/page_table.rs

```diff
@@ -185,11 +185,18 @@ impl<'a, T: PagingMode> LockedPte<'a, T> {
     ///
     /// The caller must guarantee that `paddr` references a page uniquely owned by the root
     /// `StageTwoPageTable`.
-    unsafe fn map_leaf(self, paddr: SupervisorPageAddr, perms: PteLeafPerms) -> LeafPte<'a, T> {
+    unsafe fn map_leaf(
+        self,
+        paddr: SupervisorPageAddr,
+        perms: PteLeafPerms,
+        user: bool,
```

Comment:
Nit: If it's not overkill, consider using Supervisor / User enum instead of bool.

Reply:
+1, or make it part of `PteLeafPerms`?